### PR TITLE
VideoPress + shortcode: Add support for preload/preloadcontent attribute

### PR DESCRIPTION
**Pull request by jgcaruso**
This PR adds support for the `preloadContent` player attribute to the `videopress` and `wpvideo` shortcode. I'm not sure how this has been missing for so long considering how long it has been supported in the player. I vaguely recall it existing at some point, but maybe some old VideoPress code was removed.

Fixes # https://github.com/Automattic/greenhouse/issues/1554

## Proposed changes:

* Adds support for the `preload` (or `preloadcontent`) video attribute to the `videopress` (and `wpvideo` when videopress is enabled)

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

Nope

## Testing instructions:

* Test on a jetpack test site with VideoPress enabled
* add a VideoPress shortcode to your site, include the `preload` attribute (ex `[wpvideo gSmUdNwi preload=none]`)
* View the public page, inspect the video. You should see the `preload=none` attribute on the `<video>` tag inside the iframe
* change the `preload` attribute to `preloadcontent`, publish the change
* view the public page, inspect the video. You should still see the `preload=none` attribute on the video
* Do the above test plan separately twice, with either the `Jetpack` plugin activated or the `Jetpack VideoPress` plugin activated

### CODE REVIEW

**Comment by roundhill** [comment]
Hmm do I need to have a feature flag enabled or something? My test install is using this code to render the shortcode:

https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/videopress/shortcode.php#L56

**Comment by jgcaruso** [comment]
> Hmm do I need to have a feature flag enabled or something? My test install is using this code to render the shortcode:
> 
> https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/videopress/shortcode.php#L56

Oh nice catch. I'm not sure what is going on there. At the very least I can add the same support there as well.


EDIT: Looks like its because I was using the Jetpack VideoPress plugin and not the full Jetpack plugin. If you enable Jetpack VideoPress, then disable Jetpack plugin you should be able to test the original flow. But I've added support to the Jetpack Plugin shortcode file as well. I wonder if the VideoPress plugin version will replace the original one at some point?

File: projects/packages/videopress/src/class-block-editor-content.php [hunk 1]
```diff
@@ -95,6 +100,7 @@ public static function videopress_embed_shortcode( $atts ) {
 			'controls'        => $atts['controls'],
 			'playsinline'     => $atts['playsinline'],
 			'useAverageColor' => $atts['useaveragecolor'], // The casing is intentional, shortcode params are lowercase, but player expects useAverageColor
+			'preloadContent'  => $atts['preloadcontent'], // The casing is intentional, shortcode params are lowercase, but player expects useAverageColor
```
Comment (roundhill) [inline projects/packages/videopress/src/class-block-editor-content.php]: The copied comment here is out of context.

**Review by roundhill — CHANGES_REQUESTED** [review]
Tested fine, just need to update a comment

File: projects/packages/videopress/src/class-block-editor-content.php (hunk 1 above)
Comment (jgcaruso) [reply projects/packages/videopress/src/class-block-editor-content.php]: Nice catch 😅

**Review by roundhill — DISMISSED** [review]
Looking good to me!

File: projects/packages/videopress/src/class-block-editor-content.php (hunk 1 above)
Comment (zinigor) [reply projects/packages/videopress/src/class-block-editor-content.php]: I hate to nitpick, but the comment still seems to be incorrect.

Comment (jgcaruso) [reply projects/packages/videopress/src/class-block-editor-content.php]: Sorry about that, looks like I missed one.

**Review by zinigor — DISMISSED** [review]

**Comment by jgcaruso** [comment]
@zinigor I pushed a small change to fix a failing test, but there is 1 left that is failing `Linting / Changelogger validity (pull_request)` but I don't understand why.

Do you have any insight, or is it OK to ignore it and merge?

**Review by retrofox — APPROVED** [review]

**Pull request merged**